cdisc_rules_engine/utilities/data_processor.py

```diff
@@ -262,121 +200,184 @@ def valid_meddra_term_references(self, dataframe, target, domain, **kwargs):
             MedDRAVariables.DECOD.value,
             MedDRAVariables.LLT.value,
         ]
-        code_strings = [f"{domain}{variable}" for variable in code_variables]
-        cache_key = f"meddra_valid_term_hierarchies_{dictionaries_path}"
+        code_strings = [f"{params.domain}{variable}" for variable in code_variables]
+        cache_key = f"meddra_valid_term_hierarchies_{params.meddra_path}"
         valid_term_hierarchies = self.cache.get(cache_key)
         if not valid_term_hierarchies:
-            terms: dict = self.cache.get(dictionaries_path)
+            terms: dict = self.cache.get(params.meddra_path)
             valid_term_hierarchies = MedDRATerm.get_term_hierarchies(terms)
             self.cache.add(cache_key, valid_term_hierarchies)
         column = str(uuid4()) + "_terms"
-        dataframe[column] = dataframe[code_strings].agg("/".join, axis=1)
-        result = dataframe[column].isin(valid_term_hierarchies)
+        params.dataframe[column] = params.dataframe[code_strings].agg("/".join, axis=1)
+        result = params.dataframe[column].isin(valid_term_hierarchies)
         return result
 
-    def valid_meddra_code_term_pairs(self, dataframe, target, domain, **kwargs):
-        dictionaries_path: str = kwargs.get("meddra_path")
-        if not dictionaries_path:
+    def valid_meddra_code_term_pairs(self, params: OperationParams) -> pd.Series:
```

Comment:
Currently, some method names in data processor are the actual operation names in the rules which are currently used since we are using getattr approach. Changing the function name will also lead to changing the rules which is a major change.